### PR TITLE
Detailed secret management features

### PR DESCRIPTION
**Pull request by smendis-scottlogic**
This is enhancement of secret management features submitted by kamran

### CODE REVIEW

**Review by damienjburks — DISMISSED** [review]

**Review by mlysaght2017 — APPROVED** [review]

File: services/crypto/secrets/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    threat_model_author: None
+    red_team: None
+    red_team_exercise_url: None
+    release_manager:
+      name: Damien Burks
+      github_id: damienjburks
+      company: Citi
+      summary: Initial release
+    change_log:
+      - "Test"
```
Comment (sshiells-scottlogic) [inline services/crypto/secrets/metadata.yaml:21]: Should we have "Test" here?

Comment (smendis-scottlogic) [reply services/crypto/secrets/metadata.yaml:21]: All existing metadata.yml file have this "Test" I think after Damien did some changes.

**Review by sshiells-scottlogic — APPROVED** [review]

**Comment by smendis-scottlogic** [comment]
Hi @damienjburks, can you re-approve this please? I had to rebase.

**Comment by damienjburks** [comment]
Fix the YAML lint issue @smendis-scottlogic, and I'll approve.

File: services/crypto/secrets/features.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    description: |
+      Supports automatic rotation of secrets based on a defined schedule or triggers to enhance security.
+  - id: CCC.SecMgmt.F08 # Secret Replication Policies
+    title: Secret Replication Policies
+    description: |
+      Allows configuration of secret replication policies to control replication of secrets, supporting compliance with data residency requirements.
+  - id: CCC.SecMgmt.F09 # Secure Secret Retrieval
+    title: Secure Secret Retrieval
+    description: |
+      Offers a secure API and SDK access for retrieving secrets, ensuring that secrets are transmitted securely to authorized clients.
```
Comment (eddie-knight) [inline services/crypto/secrets/features.yaml:51]:
```suggestion
      Offers a secure API and SDK access for retrieving
      secrets, ensuring that secrets are transmitted
      securely to authorized clients.
```

File: services/crypto/secrets/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+title: Secret Management
+id: CCC.SecMgmt
+description: |
+  Secret Management is a tool provided by cloud service providers
```
Comment (eddie-knight) [inline services/crypto/secrets/metadata.yaml:4]:
```suggestion
  Secret Management is a tool provided by cloud service providers
```

**Review by eddie-knight — COMMENTED** [review]
suggested changes for yaml lint

**Comment by smendis-scottlogic** [comment]
@eddie-knight thanks! @damienjburks back to you

**Review by eddie-knight — APPROVED** [review]

**Review by sshiells-scottlogic — APPROVED** [review]

**Pull request merged**